lib/ExecutionEngine/MemRefUtils.cpp

```diff
@@ -59,7 +62,18 @@ allocMemRefDescriptor(Type type, bool allocateData = true,
   auto shape = memRefType.getShape();
   int64_t size = std::accumulate(shape.begin(), shape.end(), 1,
                                  std::multiplies<int64_t>());
-  descriptor->data = reinterpret_cast<float *>(malloc(sizeof(float) * size));
+  // Align vector of f32 to the vector size boundary (to the closest greater
+  // power of two if the former isn't a power of two).
+  if (vectorType) {
+    int64_t numElements = vectorType.getNumElements();
+    size *=  numElements;
+    size_t alignment = llvm::PowerOf2Ceil(numElements * sizeof(float));
+    posix_memalign(reinterpret_cast<void **>(&descriptor->data), alignment,
+                   size * sizeof(float));
```

Comment:
Does it work on windows?

Reply:
> Does it work on windows?

I don't have a setup to test this. But I hope someone who has MLIR working on Windows could confirm. There are other alternatives and we could use malloc and shift to align but the code in context becomes relatively messier since we'd have to keep track of it for freeing the memref's.

Reply:
You won't be able to test it because the LLVM jit doesn't support windows right now, but I can confirm that posix_memalign is not available on windows. The closest thing would be _aligned_malloc.

Reply:
> I don't have a setup to test this. But I hope someone who has MLIR working on Windows could confirm

I kicked the Kokoro builds after asking :)

The windows build failed actually.

Reply:
I assume you have access to the log from the results below, but here it is:

```

7297
       "T:\src\build\projects\mlir\test\check-mlir.vcxproj" (default target) (1) ->
7298
       "T:\src\build\projects\mlir\tools\mlir-cpu-runner\mlir-cpu-runner.vcxproj" (default target) (44) ->
7299
       "T:\src\build\projects\mlir\lib\ExecutionEngine\MLIRExecutionEngine.vcxproj" (default target) (98) ->
7300
       (ClCompile target) ->
7301
         t:\src\github\llvm\llvm\projects\mlir\lib\executionengine\memrefutils.cpp(71): error C3861: 'posix_memalign': identifier not found [T:\src\build\projects\mlir\lib\ExecutionEngine\MLIRExecutionEngine.vcxproj]
7302
7303
```

Reply:
Thanks! Would there be a way to write conditionally compiled code so that _aligned_malloc is used for a Windows build? (I didn't see any such examples in MLIR.)